lib/cli.ex

```diff
@@ -4,16 +4,26 @@ defmodule Onigumo.CLI do
   }
 
   def main(argv) do
-    parsed = OptionParser.parse(argv, aliases: [C: :working_dir], strict: [working_dir: :string])
+    parsed =
+      OptionParser.parse(
+        argv,
+        aliases: [h: :help, C: :working_dir],
+        strict: [help: :boolean, working_dir: :string]
+      )
 
-    with {switches, [component], []} <- parsed,
-         {:ok, module} <- Map.fetch(@components, String.to_atom(component)) do
-      working_dir = Keyword.get(switches, :working_dir, File.cwd!())
-      module.main(working_dir)
+    with {[help: true], [], []} <- parsed do
```

Comment:
I am afraid this is not the right use case for `with`. It only collects a single value and the `else` part only contains a single branch. It’s just a pompous `if`/`else`. `case` might be more appropriate here. See  #256.

```elixir
case parsed:
  [help: true], [], []} ->
    usage_message()

  {switches, [component], []} ->
    …

  {_, _, [_ | _]} ->
    usage_message()

  {_, argv, _} when length(argv) != 1 →
    usage_message()
end
```